tests/postgres_tests/test_operations.py

```diff
@@ -581,3 +589,20 @@ def test_validate(self):
         self.assertEqual(name, "ValidateConstraint")
         self.assertEqual(args, [])
         self.assertEqual(kwargs, {"model_name": "Pony", "name": constraint_name})
+
+
+@unittest.skipUnless(connection.vendor == "postgresql", "PostgreSQL specific tests.")
+@modify_settings(INSTALLED_APPS={"append": "migrations"})
+class TestDateExtractSQL(PostgreSQLTestCase):
+    app_label = "test_date_extract_sql"
+
+    def test_date_extraction(self):
+        r = Room.objects.create(number=1)
+        hr = HotelReservation.objects.create(
+            room=r,
+            datespan=DateRange(lower="2022-2-21", upper="2022-2-22"),
+            start=datetime.datetime(2022, 2, 21, 2, 36, 10, 10000),
+            end=datetime.datetime(2022, 2, 21, 2, 36, 10, 20000),
+        )
+        qs = HotelReservation.objects.filter(start__second=F("end__second"))
+        self.assertSequenceEqual(qs, [hr])
```

Comment:
This issue is PostgreSQL-specific, but should work on all builtin backends. I'd remove these tests and add a database agnostic test to the `tests.db_functions.datetime.test_extract_trunc`, e.g.
```diff
diff --git a/tests/db_functions/datetime/test_extract_trunc.py b/tests/db_functions/datetime/test_extract_trunc.py
index 67a4b81766..682f78f149 100644
--- a/tests/db_functions/datetime/test_extract_trunc.py
+++ b/tests/db_functions/datetime/test_extract_trunc.py
@@ -899,6 +899,20 @@ class DateFunctionTests(TestCase):
             2,
         )
 
+    def test_extract_second_func_no_fractional(self):
+        start_datetime = datetime(2015, 6, 15, 14, 30, 50, 321)
+        end_datetime = datetime(2016, 6, 15, 14, 30, 50, 783)
+        if settings.USE_TZ:
+            start_datetime = timezone.make_aware(start_datetime)
+            end_datetime = timezone.make_aware(end_datetime)
+        obj = self.create_model(start_datetime, end_datetime)
+        self.assertSequenceEqual(
+            DTModel.objects.filter(
+                start_datetime__second=F("end_datetime__second"),
+            ),
+            [obj],
+        )
+
     def test_trunc_func(self):
         start_datetime = datetime(999, 6, 15, 14, 30, 50, 321)
         end_datetime = datetime(2016, 6, 15, 14, 10, 50, 123)
```